### DISCUSSION TITLE
looking for a clean way to get out of an slimmed-down existing tailwind config, back to default

### DISCUSSION THREAD

**Discussion by nbaleli-w** [tailwindlabs/tailwindcss#3581] [Help]
Hi!
when I started with tailwind, I slimmed down the config (to save bundle size) - that turned out to be a not-so-good idea, 
I learned to see that the defaults + PostCSS is the better approach.

Im now looking for a way to revert back to the defaults while keeping backward compat.

my thoughts:
1. have a new prefix from now on, to separate from the old one
2. generate a static css file containing all the used tailwind rules, and loading it along with my app statically (that way old rules will not break

I'd like to know if there's a better approach for this?
also, is there a good way (besides manually extracting the rules) to generate a static file after the postcss step?

and ofcourse kudos to you on this amazing piece of software! :)

**Comment by simonswiss** [comment]
Hey @nbaleli-w 👋

Both these approaches (prefix for a portion of your classes, or loading a "legacy" CSS file) seem like they would bring some confusion/complexity in your project. That said, I am not sure what the best solution is for your specific case.

Option 1 adds friction on the developer side, option 2 adds it on the user side.

Mind sharing what your slimmed down Tailwind config file looks like?

Also, not sure what you mean by 

> also, is there a good way (besides manually extracting the rules) to generate a static file after the postcss step?

Assuming you mean Purge CSS and not PostCSS, the output of that step _is_ a static file. Mind clarifying this question a little bit?

**Comment by nbaleli-w** [comment]
hi @simonswiss, le tme try and clarify:

becuase I changed the default spacing in the config, the rule `tw-m-t-1` will have a different meaning if I go back to the defaults (currently 1 is `8px`)

I was thinking to extract all currently used rules to static file, so that the `tw-m-t-1` rule will live on and wouldn't need to change,


> Assuming you mean Purge CSS and not PostCSS, the output of that step is a static file

Im looking for a way to extract only the tailwind rules that are currently used into their own file

> Mind sharing what your slimmed down Tailwind config file looks like?

sure, here it is 
[tailwind.config.txt](https://github.com/tailwindlabs/tailwindcss/files/5980169/tailwind.config.txt)

**Reply by NitsanBaleli** [reply]
@simonswiss if you'd be so kind and take a look at my clarification
